homeassistant/components/modbus/base_platform.py

```diff
@@ -182,12 +182,25 @@ def __init__(self, hass: HomeAssistant, hub: ModbusHub, config: dict) -> None:
         self._data_type = config[CONF_DATA_TYPE]
         self._structure: str = config[CONF_STRUCTURE]
         self._scale = config[CONF_SCALE]
-        self._precision = config.get(CONF_PRECISION, 2 if self._scale < 1 else 0)
+        self._precision = config.get(CONF_PRECISION, 2)
         self._offset = config[CONF_OFFSET]
         self._slave_count = config.get(CONF_SLAVE_COUNT, None) or config.get(
             CONF_VIRTUAL_COUNT, 0
         )
         self._slave_size = self._count = config[CONF_COUNT]
+        self._value_is_int: bool = self._data_type in (
+            DataType.INT16,
+            DataType.INT32,
+            DataType.INT64,
+            DataType.UINT16,
+            DataType.UINT32,
+            DataType.UINT64,
+        )
+        if self._value_is_int:
+            if self._min_value:
+                self._min_value = int(round(self._min_value))
+            if self._max_value:
+                self._max_value = int(round(self._max_value))
```

Comment:
```suggestion
                self._max_value = round(self._max_value)
```

Round returns an `int` by default.

Reply:
round() returns a float without decimals we want an integer.

https://www.w3schools.com/python/ref_func_round.asp

Reply:
Done